File: banchus-app/src/main/java/pe/nanamochi/banchus/protocol/handlers/PrivateMessageHandler.java
package pe.nanamochi.banchus.protocol.handlers;

import java.io.ByteArrayOutputStream;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;
import pe.nanamochi.banchus.core.Packets;
import pe.nanamochi.banchus.database.entity.Session;
import pe.nanamochi.banchus.packets.client.PrivateMessagePacket;
import pe.nanamochi.banchus.protocol.AbstractPacketHandler;
import pe.nanamochi.banchus.protocol.HandleClientPacket;

@Slf4j
@Component
@HandleClientPacket(value = Packets.OSU_PRIVATE_MESSAGE, checkForRestriction = true)
public class PrivateMessageHandler extends AbstractPacketHandler<PrivateMessagePacket> {
  @Override
  public void handle(
      PrivateMessagePacket packet, Session session, ByteArrayOutputStream responseStream) {}
}

File: banchus-app/src/main/kotlin/pe/nanamochi/banchus/protocol/handlers/PrivateMessageHandler.kt
package pe.nanamochi.banchus.protocol.handlers

import java.io.ByteArrayOutputStream
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component
import pe.nanamochi.banchus.core.PacketType
import pe.nanamochi.banchus.database.entity.Session
import pe.nanamochi.banchus.infrastructure.protocol.HandleClientPacket
import pe.nanamochi.banchus.packets.client.PrivateMessagePacket
import pe.nanamochi.banchus.protocol.AbstractPacketHandler

@Component
@HandleClientPacket(type = PacketType.OSU_PRIVATE_MESSAGE, checkForRestriction = true)
class PrivateMessageHandler :
    AbstractPacketHandler<PrivateMessagePacket>(PacketType.OSU_PRIVATE_MESSAGE) {

    private val log = LoggerFactory.getLogger(javaClass)

    override fun handle(
        packet: PrivateMessagePacket,
        session: Session,
        responseStream: ByteArrayOutputStream,
    ) {}
}
